server/routes/contacts/manage/prisoner-search/prisonerSearchController.test.ts

```diff
@@ -0,0 +1,97 @@
+import request from 'supertest'
+import { SessionData } from 'express-session'
+import { v4 as uuidv4 } from 'uuid'
+import { appWithAllRoutes, flashProvider, user } from '../../../testutils/appSetup'
+import AuditService, { Page } from '../../../../services/auditService'
+import { ENTER_TWO_CHARS_MIN } from './prisonerSearchSchema'
+
+jest.mock('../../../../services/auditService')
+
+const auditService = new AuditService(null) as jest.Mocked<AuditService>
+
+let app: Express
+let session: Partial<SessionData>
+const journeyId: string = uuidv4()
+
+beforeEach(() => {
+  app = appWithAllRoutes({
+    services: {
+      auditService,
+    },
+    userSupplier: () => user,
+    sessionReceiver: (receivedSession: Partial<SessionData>) => {
+      session = receivedSession
+      session.manageContactsJourneys = {}
+      session.manageContactsJourneys[journeyId] = {
+        id: journeyId,
+        lastTouched: new Date(),
+      }
+    },
+  })
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('GET /contacts/manage/prisoner-search', () => {
+  it('should render the search page', async () => {
+    auditService.logPageView.mockResolvedValue(null)
+
+    const response = await request(app).get(`/contacts/manage/prisoner-search/${journeyId}`)
+
+    expect(response.status).toEqual(200)
+    expect(response.text).toContain('Contacts')
+    expect(response.text).toContain('Hmpps Contacts Ui')
+    expect(auditService.logPageView).toHaveBeenCalledWith(Page.PRISONER_SEARCH_PAGE, {
+      who: user.username,
+      correlationId: expect.any(String),
+    })
+  })
+
+  it('should return to start if the journey ID is not recognised in the session', async () => {
+    await request(app)
+      .get(`/contacts/manage/prisoner-search/${uuidv4()}`)
+      .expect(302)
+      .expect('Location', '/contacts/manage/start')
+  })
+})
+
+describe('POST /contacts/manage/prisoner-search/:journeyId', () => {
+  it('should forward to the results page if there are no validation errors', async () => {
+    flashProvider.mockReturnValue({ search: ['A1111AA'] })
+
+    await request(app)
+      .post(`/contacts/manage/prisoner-search/${journeyId}`)
+      .type('form')
+      .send({ search: 'A1111AA' })
+      .expect(302)
+      .expect('Location', `/contacts/manage/prisoner-search-results/${journeyId}`)
+
+    expect(flashProvider).not.toHaveBeenCalled()
+    expect(session.manageContactsJourneys[journeyId].search.searchTerm).toEqual('A1111AA')
+  })
+
+  it('should return to search entry page if there are validation errors', async () => {
+    flashProvider.mockReturnValue({ search: [ENTER_TWO_CHARS_MIN] })
+
+    await request(app)
+      .post(`/contacts/manage/prisoner-search/${journeyId}`)
+      .type('form')
+      .send({ search: 'A' })
+      .expect(302)
+      .expect('Location', `/contacts/manage/prisoner-search/${journeyId}`)
+
+    expect(flashProvider).toHaveBeenCalledWith('validationErrors', `{"search":["${ENTER_TWO_CHARS_MIN}"]}`)
```

Comment:
I've been debating the best way to test these rules, it's fine here as there's only one rule. I'm leaning towards:
* Test in controller that a validation failure returns to page
* Cypress tests that check the validation errors are presented correctly (field and top level errors) but not every permutation
* Unit test against the schema that covers all validation scenarios